File: app/src/main/java/com/example/stop_fgastos/StopGastosApplication.java
package com.example.stop_fgastos;

import android.app.Application;

import com.example.stop_fgastos.di.AppContainer;

public final class StopGastosApplication extends Application {
    private AppContainer container;

    @Override
    public void onCreate() {
        super.onCreate();
        container = new AppContainer(this);
    }

    public AppContainer container() {
        return container;
    }
}

File: app/src/main/java/com/example/stop_fgastos/StopGastosApplication.kt
package com.example.stop_fgastos

import android.app.Application
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions

class StopGastosApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        if (FirebaseApp.getApps(this).isEmpty()) {
            val options = FirebaseOptions.Builder()
                .setApiKey("AIzaSyCg9hI4SY9Uo1vrwdLSq-liiuGtgI0nmD8")
                .setApplicationId("1:363408500943:web:aa4d5ec09bb575fc2f2fba")
                .setProjectId("stopgastos")
                .setStorageBucket("stopgastos.firebasestorage.app")
                .setGcmSenderId("363408500943")
                .build()

            FirebaseApp.initializeApp(this, options)
        }
    }
}